simulationworkflowschema/molecular_dynamics.py

```diff
@@ -384,6 +392,145 @@ class BarostatParameters(ArchiveSection):
     )
 
 
+class Lambdas(MSection):
+    """
+    Section for storing all lambda parameters for free energy perturbation
+    """
+
+    m_def = Section(validate=False)
+
+    kind = Quantity(
+        type=MEnum(
+            "output", "coulomb", "vdw", "bonded", "restraint", "mass", "temperature"
+        ),
+        shape=[],
+        description="""
+        The type of lambda interpolation
+
+        Allowed values are:
+
+        | kind          | Description                               |
+
+        | ---------------------- | ----------------------------------------- |
+
+        | `"output"`           | Lambdas for the free energy outputs saved.
+                                    These will also act as a default in case some
+                                    relevant lambdas are not specified. |
+
+        | `"coulomb"`          | Lambdas for interpolating electrostatic interactions. |
+
+        | `"vdw"`              | Lambdas for interpolating van der Waals interactions. |
+
+        | `"bonded"`           | Lambdas for interpolating all intramolecular interactions. |
+
+        | `"restraint"`        | Lambdas for interpolating restraints. |
+
+        | `"mass"`             | Lambdas for interpolating masses. |
+
+        | `"temperature"`      | Lambdas for interpolating temperature. |
+        """,
+    )
+
+    value = Quantity(
+        type=np.float64,
+        shape=[],
+        description="""
+        List of lambdas.
+        """,
+    )
+
+
+class FreeEnergyCalculationParameters(ArchiveSection):
+    """
+    Section containing the parameters pertaining to a free energy calculation workflow
+    that interpolates between two system states (defined via the interpolation parameter lambda).
+    The parameters are stored for each molecular dynamics run separately, to be referenced
+    by the overarching workflow.
+    """
+
+    m_def = Section(validate=False)
+
+    type = Quantity(
+        type=MEnum("alchemical", "umbrella_sampling"),
+        shape=[],
+        description="""
+        Specifies the type of workflow.
+        """,
+    )
+
+    lambdas = SubSection(
+        sub_section=Lambdas.m_def,
+        description="""
+        Contains the lists of lambda values defined for the interpolation of the system.
+        """,
+        repeats=True,
+    )
+
+    lambda_index = Quantity(
+        type=int,
+        shape=[],
+        description="""
+        The index of the lambda in `lambdas` corresponding to the state of the current simulation.
+        """,
+    )
+
+    atom_indices = Quantity(
+        type=np.dtype(np.int32),
+        shape=[],
+        description="""
+        List of atom indices involved in the interpolation.
+        """,
+    )
+
+    initial_state_vdw = Quantity(
+        type=MEnum("on", "off"),
```

Comment:
for binary quantities, would it not be better if we use boolean type?

Reply:
done throughout.